--- lrzsz-0.12.20.safe/Makefile.am	1998-12-30 06:19:40.000000000 -0500
+++ lrzsz-0.12.20/Makefile.am	2004-09-12 14:40:35.121631656 -0400
@@ -83,3 +83,5 @@
 rpm: $(PR).tar.gz Specfile
 	$(srcdir)/buildrpm $(srcdir)
 
+
+ACLOCAL_AMFLAGS = -I m4
--- lrzsz-0.12.20.safe/configure.in	2004-09-12 14:39:55.161706488 -0400
+++ lrzsz-0.12.20/configure.in	2004-09-12 14:40:35.816526016 -0400
@@ -260,11 +260,11 @@
 AM_GNU_GETTEXT
+AM_GNU_GETTEXT_VERSION([0.18])
 
 AC_DEFINE_UNQUOTED(LOCALEDIR,"$prefix/$DATADIRNAME")
-AC_LINK_FILES($nls_cv_header_libgt, $nls_cv_header_intl)
 
 AC_OUTPUT([Makefile intl/Makefile lib/Makefile testsuite/Makefile \
 man/Makefile po/Makefile.in src/Makefile debian/rules Specfile systype \
 src/lrzszbug],
-[sed -e "/POTFILES =/r po/POTFILES" po/Makefile.in > po/Makefile; 
+[
 chmod +x debian/rules;
 test -z "$CONFIG_HEADERS" || echo timestamp > stamp-h])
--- lrzsz-0.12.20.safe/src/Makefile.am	1998-12-28 03:38:47.000000000 -0500
+++ lrzsz-0.12.20/src/Makefile.am	2004-09-12 14:40:53.665812512 -0400
@@ -2,12 +2,12 @@
 lrz_SOURCES=lrz.c timing.c zperr.c zreadline.c crctab.c rbsb.c zm.c protname.c tcp.c lsyslog.c canit.c
 lsz_SOURCES=lsz.c timing.c zperr.c zreadline.c crctab.c rbsb.c zm.c protname.c tcp.c lsyslog.c canit.c
 noinst_HEADERS = timing.h zglobal.h zmodem.h
-datadir = $(prefix)/@DATADIRNAME@
+datadir = $(prefix)/share
 localedir = $(datadir)/locale
 CFLAGS=@CFLAGS@
 DISTCLEAN_FILES=lrzszbug
 
-LDADD = ../lib/libzmodem.a @INTLLIBS@
+LDADD = ../lib/libzmodem.a @LIBINTL@
 AUTOMAKE_OPTIONS=ansi2knr
 EXTRA_DIST = ansi2knr.1 ansi2knr.c lrzszbug.in
 INCLUDES = -I.. -I$(srcdir) -I$(top_srcdir)/src -I../intl -I$(top_srcdir)/lib
