From 978a33fefe4d939ff4f8d1e032370af270682055 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Thu, 17 Dec 2020 22:14:28 +0100
Subject: [PATCH] codegen.py: do not print stdout

Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 gio/tests/codegen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/tests/codegen.py b/gio/tests/codegen.py
index 51de0ed..8f93ac6 100644
--- a/gio/tests/codegen.py
+++ b/gio/tests/codegen.py
@@ -250,7 +250,7 @@ class TestCodegen(unittest.TestCase):
 
         result = Result(info, out, err, subs)
 
-        print('Output:', result.out)
+        #print('Output:', result.out)
         return result
 
     def runCodegenWithInterface(self, interface_contents, *args):
