Upstream-Status: Not applicable

Index: git/libkmod/libkmod-private.h
===================================================================
<<<<<<< HEAD
--- git.orig/libkmod/libkmod-private.h	2013-08-21 10:07:51.000000000 -0700
+++ git/libkmod/libkmod-private.h	2013-08-21 14:34:04.558278849 -0700
@@ -9,6 +9,10 @@
 #include "macro.h"
 #include "libkmod.h"
=======
--- git.orig/libkmod/libkmod-private.h
+++ git/libkmod/libkmod-private.h
@@ -1,6 +1,10 @@
 #ifndef _LIBKMOD_PRIVATE_H_
 #define _LIBKMOD_PRIVATE_H_
>>>>>>> cb9658cf8ab6cf009030dcadde9dc6c54b72bddc
 
+#ifndef O_CLOEXEC
+# define O_CLOEXEC 0
+#endif
+
<<<<<<< HEAD
 static _always_inline_ _printf_format_(2, 3) void
 	kmod_log_null(struct kmod_ctx *ctx, const char *format, ...) {}
 
=======
 #include <stdbool.h>
 #include <stdio.h>
 #include <syslog.h>
>>>>>>> cb9658cf8ab6cf009030dcadde9dc6c54b72bddc
