From f7da2717bc9271253962a08ba3c4664421300e47 Mon Sep 17 00:00:00 2001
From: Choong YinThong <yin.thong.choong@intel.com>
Date: Thu, 27 Jul 2017 16:39:59 -0700
Subject: [PATCH] remove uclibc

Signed-off-by: Choong YinThong <yin.thong.choong@intel.com>
---
 config.sub | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config.sub b/config.sub
index f018151..7c94b4b 100755
--- a/config.sub
+++ b/config.sub
@@ -116,7 +116,7 @@ esac
 maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
   nto-qnx* | linux-gnu* | linux-android* | linux-dietlibc | linux-newlib* | \
-  linux-musl* | linux-uclibc* | uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | \
+  linux-musl* | uclinux-gnu* | kfreebsd*-gnu* | \
   knetbsd*-gnu* | netbsd*-gnu* | netbsd*-eabi* | \
   kopensolaris*-gnu* | \
   storm-chaos* | os2-emx* | rtmk-nova*)
@@ -1391,7 +1391,7 @@ case $os in
 	      | -chorusos* | -chorusrdb* | -cegcc* \
 	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -mingw64* | -linux-gnu* | -linux-android* \
-	      | -linux-newlib* | -linux-musl* | -linux-uclibc* \
+	      | -linux-newlib* | -linux-musl* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* | -moxiebox* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
-- 
2.7.4

