Upstream-Status: Inappropriate

Index: findutils-4.9.0/gl/m4/largefile.m4
===================================================================
--- findutils-4.9.0.orig/m4/largefile.m4
+++ findutils-4.9.0/m4/largefile.m4
@@ -26,7 +26,7 @@ AC_DEFUN([gl_SET_LARGEFILE_SOURCE],
 # with _TIME_BITS.  Also, work around a problem in autoconf <= 2.69:
 # AC_SYS_LARGEFILE does not configure for large inodes on Mac OS X 10.5,
 # or configures them incorrectly in some cases.
-m4_version_prereq([2.70], [], [
+m4_version_prereq([2.73], [], [
 
 # _AC_SYS_LARGEFILE_TEST_INCLUDES
 # -------------------------------
