diff --git a/pseudo_client.c b/pseudo_client.c
index f846d54..c4c3188 100644
--- a/pseudo_client.c
+++ b/pseudo_client.c
@@ -930,6 +930,7 @@ pseudo_client_path_set(int fd, const char *path, char ***patharray, int *len) {
 
 static void
 pseudo_client_path(int fd, const char *path) {
+	pseudo_debug(PDBG_PATH, "stashing %d=%s\n", fd, path);
 	pseudo_client_path_set(fd, path, &fd_paths, &nfds);
 }
 
@@ -1609,6 +1610,7 @@ pseudo_client_op(pseudo_op_t op, int access, int fd, int dirfd, const char *path
 	/* disable wrappers */
 	pseudo_antimagic();
 
+	pseudo_debug(PDBG_CLIENT, "OP %s FD %d DIR_FD %d PATH %s\n", pseudo_op_name(op), fd, dirfd, path);
 	/* note: I am not entirely sure this should happen even if no
 	 * messages have actually been sent. */
 	if (!sent_messages) {
@@ -1629,7 +1631,14 @@ pseudo_client_op(pseudo_op_t op, int access, int fd, int dirfd, const char *path
 	if (op != OP_CHROOT && op != OP_CHDIR && op != OP_CLOSE && op != OP_DUP
 			&& pseudo_client_ignore_path_chroot(path, 0)) {
 		if (op == OP_OPEN) {
-			pseudo_client_path(fd, path);
+			int pathlen = strlen(path);
+			if (pathlen > 2 && (path[pathlen - 1]) == '/') {
+				path_extra_1 = strndup(path, pathlen-1);
+				pseudo_client_path(fd, path_extra_1);
+				free(path_extra_1);
+			} else {
+				pseudo_client_path(fd, path);
+			}
 		}
 		/* reenable wrappers */
 		pseudo_magic();
