diff --git a/aclocal.m4 b/aclocal.m4
index c2c3ca4c..4b04c53d 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -51,7 +51,7 @@ dnl ====================
 dnl serial AM1
 dnl
 dnl From Bruno Haible.
-AC_DEFUN([AM_LANGINFO_CODESET],
+AC_DEFUN([CF_LANGINFO_CODESET],
 [
 AC_CACHE_CHECK([for nl_langinfo and CODESET], am_cv_langinfo_codeset,
 	[AC_TRY_LINK([#include <langinfo.h>],
diff --git a/configure.in b/configure.in
index b71ad8ef..3f58682c 100644
--- a/configure.in
+++ b/configure.in
@@ -924,7 +924,7 @@ CPPFLAGS_after_XOPEN="$CPPFLAGS"
 CF_SIGWINCH
 
 # Checks for CODESET support.
-AM_LANGINFO_CODESET
+CF_LANGINFO_CODESET
 
 # use these variables to work around a defect in gcc's fixincludes.
 NCURSES_OK_WCHAR_T=
