From 20bc041ee2c0c7ff28aab3b8a0e486f24ac409ed Mon Sep 17 00:00:00 2001
From: Tim Orling <tim.orling@konsulko.com>
Date: Sat, 22 Apr 2023 11:22:32 -0700
Subject: [PATCH 2/2] Cargo.toml: edition 2018 -> 2021

Upstream-Status: Pending

Signed-off-by: Tim Orling <tim.orling@konsulko.com>
---
 src/rust/Cargo.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/rust/Cargo.toml b/src/rust/Cargo.toml
index 072f07bd4..5e47539b0 100644
--- a/src/rust/Cargo.toml
+++ b/src/rust/Cargo.toml
@@ -2,10 +2,10 @@
 name = "cryptography-rust"
 version = "0.1.0"
 authors = ["The cryptography developers <cryptography-dev@python.org>"]
-edition = "2018"
+edition = "2021"
 publish = false
 # This specifies the MSRV
-rust-version = "1.48.0"
+rust-version = "1.56.0"
 
 [dependencies]
 once_cell = "1"
-- 
2.34.1

