diff --git a/meta/classes-recipe/kernel-yocto.bbclass b/meta/classes-recipe/kernel-yocto.bbclass
index 1ed9887a25d..f3c350da785 100644
--- a/meta/classes-recipe/kernel-yocto.bbclass
+++ b/meta/classes-recipe/kernel-yocto.bbclass
@@ -4,10 +4,7 @@
 # SPDX-License-Identifier: MIT
 #
 
-python () {
-    if bb.utils.contains('DISTRO_FEATURES', 'rust-kernel', True, False, d):
-        d.appendVar('inherit', ' rust-common')
-}
+INHERIT:append = " ${@bb.utils.contains('DISTRO_FEATURES', 'rust-kernel', ' rust-common', '', d)}"
 
 # remove tasks that modify the source tree in case externalsrc is inherited
 SRCTREECOVEREDTASKS += "do_validate_branches do_kernel_configcheck do_kernel_checkout do_fetch do_unpack do_patch"
@@ -32,10 +29,10 @@ KMETA_AUDIT ?= "yes"
 KMETA_AUDIT_WERROR ?= ""
 KMETA_CONFIG_FEATURES ?= ""
 
-RUST_DEBUG_REMAP = "${@bb.utils.contains('DISTRO_FEATURES', 'rust-kernel', '--remap-path-prefix=${WORKDIR}=.\
-                    --remap-path-prefix=${TMPDIR}/work-shared=.',"",d)}"
+RUST_DEBUG_REMAP = "${@bb.utils.contains('DISTRO_FEATURES', 'rust-kernel', '--remap-path-prefix=${WORKDIR}=${TARGET_DBGSRC_DIR} \
+                      --remap-path-prefix=${TMPDIR}/work-shared=${TARGET_DBGSRC_DIR}', '',d)}"
 KRUSTFLAGS:append = "${RUST_DEBUG_REMAP}"
-EXTRA_OEMAKE:append = " KRUSTFLAGS='${KRUSTFLAGS}' "
+EXTRA_OEMAKE:append = "${@bb.utils.contains('DISTRO_FEATURES', 'rust-kernel', ' KRUSTFLAGS='${KRUSTFLAGS}'', '',d)}"
 
 # returns local (absolute) path names for all valid patches in the
 # src_uri
