Index: openocd-0.7.0/config_subdir.m4
===================================================================
--- openocd-0.7.0.orig/config_subdir.m4
+++ openocd-0.7.0/config_subdir.m4
@@ -22,6 +22,7 @@ AC_CONFIG_SUBDIRS([$1])
 
 m4_ifblank([$2], [rm -f $srcdir/$1/configure.gnu],
 [printf "#!/bin/sh
-"\$"SHELL "../$srcdir/$1/configure" $2 \""\$"@"\" > "$srcdir/$1/configure.gnu"
+"\$"SHELL "../$srcdir/$1/configure.sh" $2 \""\$"@"\" > "$srcdir/$1/configure.gnu"
+printf \#\!"/bin/sh\ndir="\"\`"dirname "\"\$"0"\"\`"/autosetup"\""\nWRAPPER="\"\$"0"\""; export WRAPPER; exec "\"\`\$"dir/find-tclsh"\`\"" "\"\$"dir/autosetup"\"" "\"\$"@"\""\n" > "$srcdir/$1/configure.sh"
 ])
 ])
